indico/modules/events/api.py

```diff
@@ -255,8 +273,141 @@ def _iterate_objs(objIds):
                 obj = ch.getById(objId, True)
                 if obj is not None:
                     yield obj
+        return self.serialize_events(events)
 
-        return self._process(_iterate_objs(idlist))
+    def serialize_events(self, events):
+        return map(self._build_event_api_data, events)
+
+    def _serialize_date(self, date):
+        if date:
+            return {
+                'date': str(date.date()),
+                'time': str(date.time()),
+                'tz': str(date.tzinfo)
+            }
+
+    def _serialize_persons(self, persons, person_type):
+        result = []
+        for person in persons:
+            result.append(self._serialize_person(person, person_type))
+        return result
+
+    def _serialize_person(self, person, person_type):
```

Comment:
You sometimes pass `User` objects here and sometimes person links. person links do not have an email (since that's only stored on the person itself) - you either need to add an `email` property to `PersonLinkBase` or get the email from the actual person instead of the person link. the property is probably cleaner...